Feature bracketed paste on terminal

erw7 opened the pull request:
Ref #11418.

erw7 committed:
Add support for bracketed paste mode in terminal windows

erw7 committed:
Change to filter control characters when pasting a terminal window

Change to filter the following control characters in the same way as
xterm.

\x00-\x07, \x0b-\x0c, \x0e-\x1f

erw7 committed:
Add termpastefilter option

Change to specify a character to be filtered as an option when pasting on
the terminal.

mckelly2833 commented:
After updating a few lines of a merge conflict in src/nvim/option.c this works great.
Thank you @erw7 
Someone please merge when possible :-)

erw7 commented:
I've fixed the conflicts. For my part, I'd like to hear the opinions of other committers on the validity of the options, etc. before merging this one.

jamessan commented on runtime/doc/options.txt:
```diff
@@ -6280,6 +6280,29 @@ A jump table for the options with a short description can be found at |Q_op|.
 	Requires an ISO-8613-3 compatible terminal.
 
+						*'termpastefilter'* *'tpf'*
+'termpastefilter' 'tpf'	string	(default: "")
```
Why do we need this option?

erw7 replied:
This is to prevent problems such as `\x1b[201~` being included in the text to be pasted, which would cause the parenthesized paste mode to complete. As far as I know, `xterm` and `mintty` have such an option. Also, although I couldn't find the option in the manual, `tmux` also converts `\x1b` to `^[` when pasting.  

#### man xterm

```
disallowedPasteControls (class DisallowedPasteControls)
               The allowPasteControls resource is normally used to prevent pasting C1 controls, as well as non-formatting C0 controls such as the ASCII
               escape character.  Those characters are simply ignored.  This resource further extends the set of control characters which cannot be pasted,
               converting each into a space.
```

#### man mintty

```
Filtering pasted text (FilterPasteControls=)
         With this setting, pasted text can be filtered for selected control characters which are then replaced by space. Filtering is not applied when
         Control is held while pasting. The option is a comma-separated list of character tags:
```

mckellygit commented:
Thanks again for this great PR.
Is there anything we can do to help get it reviewed/approved/merged ?
thx,
-m

jamessan commented:
Assuming CI passes, this looks good to merge.

erw7 committed:
Change default for tpf option

Change the default of tpf to the same value as the default of xterm's
disallowedPasteControls, because it may be a security risk.

erw7 commented:
I'm sorry I haven't had much time lately. Changed the default for the `tpf` option, as it could be a security issue. If everything looks good, I'll merge them in a few days.

mckellygit commented:
Thanks again,
-m
